ibc-testkit/src/testapp/ibc/core/types.rs

```diff
@@ -86,6 +87,8 @@ where
     pub packet_ack_store: BinStore<SharedStore<S>, AckPath, AcknowledgementCommitment>,
     /// Map of host consensus states
     pub host_consensus_states: Arc<Mutex<BTreeMap<u64, AnyConsensusState>>>,
+    /// Map of older ibc commitment proofs
+    pub ibc_commiment_proofs: Arc<Mutex<BTreeMap<u64, CommitmentProof>>>,
```

Comment:
I am thinking our new mock machinery will be going to be an example of IBC module implementation, and even showcasing part of the ibc-testkit in our documentations later. Therefore, I'm unsure about adding this field because commitment proofs usually come from the host storage, not the host's module system, which is the case here too. So, we logically shouldn't need to store it again under the IBC module.

Reply:
Yeah, I see. Things differ from ibc-go.
This is a symptom of our `QueryContext` design. By making the `ProverContext` the primary trait, we are implicitly requiring the IBC module to be aware of proofs, which logically should not be the case.
Let's proceed for now as I don't see an immediate alternative. However, this issue should be addressed, as it may lead people in the wrong direction.

Reply:
Carlo's explanation regarding this:

> It should include two proofs: one for the IAVL tree of the IBC store (i.e. the proof of inclusion within the IBC store) and another one for Simple Tree (i.e. the proof of inclusion of  the IBC store in the multistore).

It seems that IBC module should be aware of IBC commitment prefix proofs. We have to do this in the same way we have `host_*` methods.

```rs
pub trait ProvableContext {
    fn get_ibc_commitment_prefix_proof_on_host(&self, height: Height) -> Option<MerkleProof>
    fn get_local_store_proof(&self, height: Height, path: &Path) -> Option<MerkleProof>;
    fn get_proof(&self, height: Height, path: &Path) -> Option<MerkleProof> {
        let commitment_prefix_proof = self.get_ibc_commitment_prefix_proof_on_host(height);
        let local_proof = get_local_store_proof(height, path);
        return make_merkle_proof_chain([commitment_prefix_proof, local_proof]);
    }
}
```

Reply:
As discussed, the architecture of Cosmos SDK chains may not directly apply to other non-Cosmos chains or rollups. Let's explore this further as a separate issue. Refactoring these parts would likely require a bit more studying and experimentation.